lib/data/models/petal.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
+import 'package:learning_compass_exp/data/models/petal_names.dart';
+
+class Petal {
+  final Color color;
+  final double progress;
+  final PetalName name;
+  final double angle;
+  final IconData icon;
+
```

Comment:
I guess out of these variables we have to keep progress at the very least in the state. Everything else is atm static, but for the sake of convenience we will probably have to keep at least the icon here as well. The static data could come from json files. 

We could also create a constants-file where we have a translation table for the icon and potentially some other stuff. It could go in like **/data/constants/*.dart file and it could be a map with an identifier to a petal as the keys.

My concern with this method is that it makes some of the data a bit all over the place; some of it in json-files, some of it in this map, some of it in the state. Another option is to have all petal related specific details (so everything except actual construct information) in a dart file as a constant. That would sorta make sense because you'd have dynamic data in the state, static implementation details in a constant, and actual static data in a json file. But is this good form? My instincts on this kinda make me shiver at the idea of what is essentially a global variable.